src/test/java/rx/internal/operators/OperatorObserveOnTest.java

```diff
@@ -523,14 +523,21 @@ public boolean hasNext() {
         assertEquals(7, generated.get());
     }
 
+    /**
+     * Tests in {@link BackpressureTests} like testObserveOnWithSlowConsumer already cover the case with default buffer
+     * size, the code path is virtually identical.
+     */
     @Test
     public void testQueueFullEmitsError() {
         final CountDownLatch latch = new CountDownLatch(1);
+        // randomize buffer size, note that underlying implementations may be tuning the real size to a power of 2
+        // which can lead to unexpected results when adding excess capacity (e.g.: see ConcurrentCircularArrayQueue)
+        final int capacity = (int)Math.pow(2, new Random().nextInt(10));
```

Comment:
I agree with @artem-zinnatullin, leave this test as it was and create a new one with a loop over some queue sizes:

```java
for (int i = 1; i <= 1024; i = i * 2) {
   final int capacity = i;
   // ...
}
```